x/gamm/keeper/migrate_test.go

```diff
@@ -308,6 +311,47 @@ func (suite *KeeperTestSuite) TestReplaceMigrationRecords() {
 			},
 			expectErr: true,
 		},
+		{
+			name: "Mismatch denom0 between the two pools",
+			testingMigrationRecords: []types.BalancerToConcentratedPoolLink{
+				{
+					BalancerPoolId: 1,
+					ClPoolId:       3,
+				},
+				{
+					BalancerPoolId: 2,
+					ClPoolId:       4,
+				},
+			},
+			overwriteBalancerDenom0: "uosmo",
+			expectErr:               true,
+		},
+		{
+			name: "Mismatch denom1 between the two pools",
+			testingMigrationRecords: []types.BalancerToConcentratedPoolLink{
+				{
+					BalancerPoolId: 1,
+					ClPoolId:       3,
+				},
+				{
+					BalancerPoolId: 2,
+					ClPoolId:       4,
+				},
+			},
+			overwriteBalancerDenom1: "uosmo",
+			expectErr:               true,
+		},
```

Comment:
would it be possible to change these test cases so that we have one link?

Reply:
https://github.com/osmosis-labs/osmosis/pull/4126/commits/25e6048ee13e479383cfef8217f0215673c3c20e